storage_service/locations/api/beta/remple/openapi.py

```diff
@@ -0,0 +1,1573 @@
+            path_dict = {
+                http_method: OrderedDict([
+                    ('operationId', operation_id),
+                    ('summary', custom_endpoint.summary),
+                    ('description', custom_endpoint.description),
+                ])
+            }
+            for attr in ('parameters', 'request_body', 'responses', 'tags'):
+                val = getattr(custom_endpoint, attr)
+                if val:
+                    path_dict[http_method][attr] = val
+            paths[path] = path_dict
+
+    def _get_search_request_body_examples(self, resource_name):
+        """Note: unfortunately, these examples will not be displayed in the
+        Swagger UI since that functionality is not implemented yet. See
+        https://github.com/swagger-api/swagger-ui/issues/3771.
+        """
+        if resource_name == 'package':
+            array_search_example = {
+                'paginator': {'page': 1, 'items_per_page': 10},
+                'query': {
+                    'filter': [
+                        'and', [['description', 'like', '%a%'],
+                                ['not', ['description', 'like', 'T%']],
+                                ['or', [['size', '<', 1000],
+                                        ['size', '>', 512]]]]]}}
+            object_search_example = {
+                'paginator': {'page': 1, 'items_per_page': 10},
+                'query': {
+                    'filter': {
+                        'conjunction': 'and',
+                        'complement': [{'attribute': 'description',
+                                        'relation': 'like',
+                                        'value': '%a%'},
+                                       {'negation': 'not',
+                                        'complement': {'attribute': 'description',
+                                                       'relation': 'like',
+                                                       'value': 'T%'}},
+                                       {'conjunction': 'or',
+                                        'complement': [{'attribute': 'size',
+                                                        'relation': '<',
+                                                        'value': 1000},
+                                                       {'attribute': 'size',
+                                                        'relation': '>',
+                                                        'value': 512}]}]}}}
+            return {'ArraySearchOverPackagesExample': array_search_example,
+                    'ObjectSearchOverPackagesExample': object_search_example}
+
+    def _set_search_paths(self, paths, resource_name, resource_cls, pk_patt,
+                          rsrc_collection_name):
+        for action in ('search_post', 'new_search'):
+            http_method = {'search_post': 'post', 'new_search': 'get'}.get(
+                action, 'search')
```

Comment:
Document or remove the implicit usage here of the non-standard HTTP method `SEARCH`.